### PR TITLE
Defined the Protobuf definitions for header, block, transaction and sending the data through the P2P layer

### PR DESCRIPTION
**Pull request by gameofpointers**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,7 +31,7 @@ mocks:
 protogen:
 	@echo "Generating protobuf files"
 	@protoc --go_out=. --go_opt=paths=source_relative \
-	./p2p/pb/*.proto 
+	./core/types/*.proto ./common/*.proto ./p2p/pb/*.proto
```
Comment (wizeguyy) [inline Makefile:34]: Can you do `./**/*.proto` instead?

Comment (gameofpointers) [reply Makefile:34]: yes, we can, will do this

**Review by wizeguyy — APPROVED** [review]

**Review by Djadih — APPROVED** [review]

File: quai/p2p_backend.go [hunk 2]
```diff
@@ -50,13 +51,33 @@ func (qbe *QuaiBackend) SetZoneApiBackend(zoneBackend quaiapi.Backend, location
...
-func (qbe *QuaiBackend) OnNewBroadcast(sourcePeer p2p.PeerID, data interface{}) bool {
-	panic("todo")
+func (qbe *QuaiBackend) OnNewBroadcast(sourcePeer p2p.PeerID, data interface{}, nodeLocation common.Location) bool {
+	switch data.(type) {
+	case types.Block:
+		block := data.(types.Block)
+		backend := *qbe.GetBackend(nodeLocation)
+		if backend == nil {
+			log.Error("no backend found")
+			return false
```
Comment (Djadih) [inline quai/p2p_backend.go:73]: Why should we return bool instead of error?

Comment (gameofpointers) [reply quai/p2p_backend.go:73]: @alejoacosta74 what was the rationale?

**Pull request merged**